SingularityUI/app/components/tasks/TasksPage.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import filterSelector from '../../selectors/tasks/filterSelector';
+import decomSelector from '../../selectors/tasks/decomSelector';
+
+import TaskFilters from './TaskFilters';
+import { FetchAction } from '../../actions/api/tasks';
+import { KillAction } from '../../actions/api/task';
+import { RunAction } from '../../actions/api/request';
+import { FetchRunAction } from '../../actions/api/request';
+import { FetchRunHistoryAction } from '../../actions/api/request';
+import { FetchAction as FetchFilesAction } from '../../actions/api/taskFiles';
+
+import UITable from '../common/table/UITable';
+import KillTaskModal from '../common/KillTaskModal';
+import RunNowModal from '../common/RunNowModal';
+import TaskLauncher from './TaskLauncher';
+import { TaskId, StartedAt, Host, Rack, CPUs, Memory, ActiveActions, NextRun, PendingType, DeployId, ScheduledActions, ScheduledTaskId, CleanupType, JSONAction, InstanceNumber } from './Columns';
+
+class TasksPage extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      filter: {
+        taskStatus: props.state,
+        requestTypes: props.requestsSubFilter == 'all' ? TaskFilters.REQUEST_TYPES : props.requestsSubFilter.split(','),
+        filterText: props.searchFilter,
+        loading: false
+      }
+    }
+  }
+
+  handleFilterChange(filter) {
+    const lastFilterTaskStatus = this.state.filter.taskStatus;
+    this.setState({
+      loading: lastFilterTaskStatus != filter.taskStatus,
+      filter: filter
+    });
+
+    const requestTypes = filter.requestTypes.length == TaskFilters.REQUEST_TYPES.length ? 'all' : filter.requestTypes.join(',');
+    this.props.updateFilters(filter.taskStatus, requestTypes, filter.filterText);
+    app.router.navigate(`/tasks/${filter.taskStatus}/${requestTypes}/${filter.filterText}`);
+
+    if (lastFilterTaskStatus != filter.taskStatus) {
+      this.props.fetchFilter(filter.taskStatus).then(() => {
+        this.setState({
+          loading: false
+        });
+      });
+    }
+  }
+
+  handleTaskKill(taskId, data) {
+    this.props.killTask(taskId, data);
+  }
+
+  handleRunNow(requestId, data) {
+    this.props.runRequest(requestId, data).then((response) => {
+      // console.log(data, response.data);
```

Comment:
I wrote a commit hook for this a while ago. It's not perfect, but it catches most of these:
https://gist.github.com/anonymous/7572da108a6e1c5ffce6064c94f85b96